crates/avalanche-types/src/warp/client.rs

```diff
@@ -0,0 +1,43 @@
+use std::io::{Error, ErrorKind, Read, Result};
+use std::str::FromStr;
+
+use crate::ids::Id;
+use crate::proto::pb::warp::{signer_client, SignRequest, SignResponse};
+use prost::bytes::Bytes;
+use tonic::transport::Channel;
+
+#[derive(Clone)]
+pub struct WarpSignerClient {
+    inner: signer_client::SignerClient<Channel>,
+}
+
+impl WarpSignerClient {
+    pub fn new(client_conn: Channel) -> Self {
+        Self {
+            inner: signer_client::SignerClient::new(client_conn)
+                .max_decoding_message_size(usize::MAX)
+                .max_encoding_message_size(usize::MAX),
+        }
+    }
+}
+
+#[tonic::async_trait]
+impl super::WarpSignerClient_ for WarpSignerClient {
+    async fn sign(
+        &self,
+        network_id: u32,
+        source_chain_id: &str,
+        payload: &[u8],
+    ) -> Result<SignResponse> {
+        let mut client = self.inner.clone();
```

Comment:
So you have to create a new signer client every time you need to sign? The clone here should be explicit by the user (or they can wrap their channel in some kind of lock). This masks the internal functionality and doesn't give users the option to do things more efficiently.